### PR TITLE
Add a feature gate to access the delete todo feature

### PR DESCRIPTION
**Pull request by lttsNilesh**

### CODE REVIEW

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt [hunk 1]
```diff
@@ -33,4 +34,13 @@ object StatsIgUtil {
...
     }
 
+    /**
+     * Feature Gate function to check if access to delete of to-do list is granted.
+     *
+     * @return Boolean value of Feature Gate of FEATURE_GATE_DELETE_TODO_ACCESS.
+     */
+    fun isDeleteTodoFeatureEnabled(): Boolean {
+        return Statsig.checkGate(FEATURE_GATE_DELETE_TODO_ACCESS)
+    }
```
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:44]: this is a lot of unnecessary code - remove the constant, remove this method, and just add `Statsig.checkGate` inline in the method

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/TopAppBars.kt [hunk 2]
```diff
@@ -161,8 +161,10 @@ fun TaskDetailTopAppBar(onBack: () -> Unit, onDelete: () -> Unit) {
             }
         },
         actions = {
-            IconButton(onClick = onDelete) {
-                Icon(Icons.Filled.Delete, stringResource(id = R.string.menu_delete_task))
+            if (StatsIgUtil.isDeleteTodoFeatureEnabled()) {
```
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/TopAppBars.kt:164]: just add `Statsig.checkGate` here

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 1 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:44]: Okay. I have removed the StatsigUtil and replace with Statsig.checkGate

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/TopAppBars.kt (hunk 2 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/TopAppBars.kt:164]: Okay. I have removed the StatsigUtil and replace with Statsig.checkGate

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 1 above)
Comment (tore-statsig) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:44]: Done in https://github.com/statsig-io/samples/pull/5

**Review by tore-statsig — COMMENTED** [review]
I think you can close this one, since https://github.com/statsig-io/samples/pull/5 has all the work in it?

**Pull request merged**